### ISSUE TITLE
FR: A more reusable way of declaring tags and acls.

### ISSUE THREAD

**Issue by samoht9277** [tailscale/tailscale#11333]
### What are you trying to do?

I have a whole bunch of servers inside my Tailnet. Each server corresponds to a different project, and each project has many employees that need access.

When I started to declare each group, I came across the realization that I have to create an _acl_ and a _tag_ for each group.

E.g.:
```
"group:number-one": [
    "user11@company.com",
    "user2@company.com",
    "user8@company.com",
    "user4@company.com",
    "user5@company.com",
    "user12@company.com"
],
"tagOwners": {
    "tag:number-one":              ["autogroup:admin"],
},
"acls": [
    {
        "action": "accept",
        "src":    ["group:number-one"],
        "dst":    ["tag:number-one:*"],
    },
]

```

Performing this task repeatedly (I currently have 13 groups in total, algonside their _tag_ and _acl_) leads to challenges in maintenance and readability.

### How should we solve this?

I believe that the should be a way to declare groups, with users inside (just like there is right now) but while having a way to create _tags_ and _acls_ in a more sophisticated manner.

E.g.:

```
"group:number-one": [
    "user11@company.com",
    "user2@company.com",
    "user8@company.com",
    "user4@company.com",
    "user5@company.com",
    "user12@company.com"
],
"tagOwners": {
    "tag:number-one":              ["autogroup:admin"],
},
"acls": [
    {
        "action": "accept",
        "src":    ["group:selfgroup"],
        "dst":    ["tag:selftag:*"],
    },
]
```

Inside the _acls_ declaration, the `selfgroup` and `selftag` mean that if a group called `number-one` exists, as well as a tag, then accept the connection. (The `self` names are just examples!)

In other words, this example means that there is no need for creating _N_ acls for _N_ given groups, given that both the tag and the acl are called the same.

PS: It would be cool to have a way of declaring tags in a shorter way, assuming the owner is `autogroup:admin` (Yes, I know that empty brackets mean `autogroup:admin`, aswell).

### What is the impact of not solving this?

Having multiple groups, thus _tags_ and _acls_ mean the the whole ACL file is way longer than it should. The example above shows a way of declaring a reusable _acl_ declaration for all groups and tags with the same name.

Currently, the only way I'm aware of working this around is declaring both _tags_ and _acls_ in the ACL file, leading to a super long file.
